eng/pipelines/templates/jobs/archetype-sdk-integration.yml

```diff
@@ -92,18 +122,40 @@ jobs:
       - script: |
           node common/scripts/install-run-rush.js install
         displayName: "Install dependencies"
+        condition: eq(variables['DependencyVersion'],'repo')
 
       # Option "-p max" ensures parallelism is set to the number of cores on all platforms, which improves build times.
       # The default on Windows is "cores - 1" (microsoft/rushstack#436).
       - script: |
           node common/scripts/install-run-rush.js build -t "${{parameters.PackageName}}" --verbose -p max
         displayName: "Build libraries"
+        condition: eq(variables['DependencyVersion'],'repo')
 
       # Option "-p max" ensures parallelism is set to the number of cores on all platforms, which improves build times.
       # The default on Windows is "cores - 1" (microsoft/rushstack#436).
       - script: |
           node common/scripts/install-run-rush.js build:test -t "${{parameters.PackageName}}" --verbose -p max
         displayName: "Build test assets"
+        condition: eq(variables['DependencyVersion'],'repo')
```

Comment:
Building on my feedback about not setting DependencyVersion in the matrix by default that would change this condition to `condition: eq(variables['DependencyVersion'],'')`